### PR TITLE
r/aws_eks_cluster: Migrate 'enabled_cluster_log_types' from list to set

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Fixes https://github.com/terraform-providers/terraform-provider-aws/issues/8279.

Acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEksCluster_Logging'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSEksCluster_Logging -timeout 120m
=== RUN   TestAccAWSEksCluster_Logging
=== PAUSE TestAccAWSEksCluster_Logging
=== CONT  TestAccAWSEksCluster_Logging
--- PASS: TestAccAWSEksCluster_Logging (1282.26s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1282.318s
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Added _enabled_cluster_log_types_ attribute to the `aws_eks_cluster` data source.
Acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEksClusterDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSEksClusterDataSource_basic -timeout 120m
=== RUN   TestAccAWSEksClusterDataSource_basic
=== PAUSE TestAccAWSEksClusterDataSource_basic
=== CONT  TestAccAWSEksClusterDataSource_basic
--- PASS: TestAccAWSEksClusterDataSource_basic (1282.34s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1282.396s
```

File: aws/resource_aws_eks_cluster.go [hunk 1]
```diff
@@ -38,6 +38,9 @@ func resourceAwsEksCluster() *schema.Resource {
 			Delete: schema.DefaultTimeout(15 * time.Minute),
 		},
 
+		SchemaVersion: 1,
+		MigrateState:  resourceAwsEksClusterMigrateState,
```
Comment (bflad) [inline aws/resource_aws_eks_cluster.go]: Hey @ewbankkit 👋 Is there a particular reason why a state migration was implemented here? `TypeList` to `TypeSet` attribute conversions should not require these as the old indexes should be overwritten during refresh. Less is sometimes more! 😉 Thanks.

**Comment by ewbankkit** [comment]
@bflad Yes, I noticed that when I did a `terraform plan` on an existing state file having changed the type from _List_ to _Set_ without adding the state migration code that no diff was detected.
I've removed the state migration code.

#### No diffs on existing state:

```console
$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

...
aws_eks_cluster.cluster: Refreshing state... (ID: xxxxxxxx)
...

------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.
```

#### Acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEksCluster_Logging'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSEksCluster_Logging -timeout 120m
=== RUN   TestAccAWSEksCluster_Logging
=== PAUSE TestAccAWSEksCluster_Logging
=== CONT  TestAccAWSEksCluster_Logging
--- PASS: TestAccAWSEksCluster_Logging (1311.92s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1311.978s
```

**Comment by nywilken** [comment]
This has been released in [version 2.8.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#280-april-26-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**